app/controllers/icons_controller.rb

```diff
@@ -87,8 +86,7 @@ def replace
     gon.gallery = Hash[all_icons.map { |i| [i.id, {url: i.url, keyword: i.keyword}] }]
     gon.gallery[''] = {url: view_context.image_path('icons/no-icon.png'), keyword: 'No Icon'}
 
-    post_ids = Reply.where(icon_id: @icon.id).select(:post_id).distinct.pluck(:post_id)
-    all_posts = Post.where(icon_id: @icon.id) + Post.where(id: post_ids)
+    all_posts = Post.where(id: Reply.where(icon_id: @icon.id).select(:post_id).distinct.pluck(:post_id))
```

Comment:
instead of `all_posts = [thing]`, remove following line and use `@posts = [thing]`, we already did distinct to dedupe for us (and Rails will dedupe duplicate ids for you anyway, `Post.where(id: [1,1,2])` returns two objects)